packages/dd-trace/src/llmobs/index.js

```diff
@@ -46,7 +52,20 @@ function enable (config) {
 
   // distributed tracing for llmobs
   injectCh.subscribe(handleLLMObsParentIdInjection)
-  telemetry.recordLLMObsEnabled(startTime, config)
+
+  setAgentStrategy(config, useAgentless => {
+    if (useAgentless && !(config.apiKey && config.site)) {
+      throw new Error(
+        'Cannot send LLM Observability data without a running agent or without both a Datadog API key and site.\n' +
+        'Ensure these configurations are set before running your application.'
+      )
+    }
+
+    evalWriter?.setAgentless(useAgentless)
+    spanWriter?.setAgentless(useAgentless)
+
+    telemetry.recordLLMObsEnabled(startTime, config)
```

Comment:
moved this telemetry metric to be inside the agent strategy callback - do we want to define our llmobs startup/enabled metric to include this async op, or just do it at the end of this function synchronously?

Reply:
I think we should probably include this async op, just because this is a pretty important part of enabling (writers)